crates/sui-graphql-rpc/src/types/dot_move/on_chain.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use async_graphql::ScalarType;
+use move_core_types::language_storage::StructTag;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use sui_types::{
+    base_types::{ObjectID, SuiAddress},
+    collection_types::VecMap,
+    dynamic_field::Field,
+    id::ID,
+    object::MoveObject as NativeMoveObject,
+};
+
+use crate::types::base64::Base64;
+
+use super::{
+    config::{MoveRegistryConfig, MOVE_REGISTRY_MODULE, MOVE_REGISTRY_TYPE},
+    error::MoveRegistryError,
+};
+
+const MAX_LABEL_LENGTH: usize = 63;
+
+/// Regex to parse a dot move name. Version is optional (defaults to latest).
+/// For versioned format, the expected format is `app@org/v1`.
+/// For an unversioned format, the expected format is `app@org`.
+///
+/// The unbound regex can be used to search matches in a type tag.
+/// Use `VERSIONED_NAME_REGEX` for parsing a single name from a str.
+const _VERSIONED_NAME_UNBOUND_REGEX: &str = concat!(
```

Comment:
Sad times about not being able to `concat!` other `const`s -- this is a good compromise though.

Reply:
Yep... I felt it...